src/wiki/plugins/redlinks/mdx/redlinks.py

```diff
@@ -65,19 +44,30 @@ def get_class(self, el):
             return
         if url.scheme == "mailto":
             return
-        if url.scheme or url.netloc or url.path.startswith("/"):
-            # Contains a hostname or is an absolute link => external
+        if url.scheme or url.netloc:
+            # Contains a hostname or url schema ⇒ External link
             return self.external_class
         # Ensure that path ends with a slash
         relpath = url.path.rstrip("/") + "/"
-        target = urljoin_internal(self.my_urlpath.path, relpath)
-        if target is None:
-            # Relative path goes outside wiki URL space => external
+        try:
+            target = resolve(urljoin(self.my_urlpath, relpath))
+        except Resolver404:
+            # Broken absolute link
+            return self.external_class
+
+        if target.app_names != ["wiki"]:
+            # Links outside wiki
             return self.external_class
+
         try:
-            URLPath.get_by_path(target)
-        except URLPath.DoesNotExist:
+            article, urlpath = which_article(**target.kwargs)
```

Comment:
- It could be an article link, instead of a link given by URLPath.
 - It could be an URLPath that exists, but has no article (the decorator/now which_article function points that option out)
 - It could be a deleted article, which would still be an existing URLPath
 - etc.

None of these are non-existant URLPaths errors, all of this should be red links.

In any case, it seemed right to use the same internal logic for “what article does this path refer to” in both the `get_article` decorator and any other function that wonders to which article a hyperlink leads (i.e. here for redlinks, and in whatlinkswhere) – that was the whole point of extracting the `which_article` function. Implement it with all its complexity once, and then re-use it levelling differences where necessary.

Reply:
I can see that you are spot-on here -- there isn't any way that this fix should scope-creep into resolving things in other ways than what is already offered and used throughout the codebase :+1: Thanks for clarifying!